File: src/test/unit/languages/javascript/fixtures/parent-calls.js
// Parent class method calls using super

class BaseService {
  constructor() {
    this.isActive = true;
  }

  validate(data) {
    return typeof data === 'object' && Object.keys(data).length > 0;
  }

  save(data) {
    console.log('BaseService: Saving', data);
    return data;
  }
}

class UserService extends BaseService {
  constructor(repository) {
    super();
    this.repository = repository;
  }

  validate(data) {
    // Call parent validation first
    if (!super.validate(data)) {
      return false;
    }
    
    // Additional validation
    return data.email !== undefined;
  }

  save(user) {
    if (this.validate(user)) {
      // Call parent save method
      const result = super.save(user);
      this.repository.add(result);
      return result;
    }
    return null;
  }
}

class AdminService extends UserService {
  constructor(repository, logger) {
    super(repository);
    this.logger = logger;
  }

  save(user) {
    this.logger.log('Saving admin user');
    // Call parent save
    return super.save(user);
  }
}

module.exports = { BaseService, UserService, AdminService };

File: packages/core/src/test/unit/languages/python/fixtures/parent_calls.py
# Parent class method calls using super()

class BaseService:
    """Base service with validation"""
    
    def __init__(self):
        self.is_active = True
    
    def validate(self, data: dict) -> bool:
        """Validate data"""
        return isinstance(data, dict) and len(data) > 0
    
    def save(self, data: dict):
        """Save data"""
        print(f"BaseService: Saving {data}")
        return data


class UserService(BaseService):
    """User service that calls parent methods"""
    
    def __init__(self, repository):
        super().__init__()  # Call parent constructor
        self.repository = repository
    
    def validate(self, data: dict) -> bool:
        """Override validate with super call"""
        # Call parent validation first
        if not super().validate(data):
            return False
        
        # Additional validation
        return 'email' in data
    
    def save(self, user: dict):
        """Override save with super call"""
        if self.validate(user):
            # Call parent save method
            result = super().save(user)
            self.repository.add(result)
            return result
        return None


class AdminService(UserService):
    """Admin service extending UserService"""
    
    def __init__(self, repository, logger):
        super().__init__(repository)  # Call parent constructor
        self.logger = logger
    
    def save(self, user: dict):
        """Override save with logging"""
        self.logger.log("Saving admin user")
        # Call parent save (which itself calls BaseService.save)
        return super().save(user)
